buildSrc/src/main/java/org/springframework/gradle/github/release/CreateGitHubReleaseTask.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.gradle.github.release;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.gradle.api.Action;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * @author Steve Riesenberg
+ */
+public class CreateGitHubReleaseTask extends DefaultTask {
+	@Input
+	private RepositoryRef repository = new RepositoryRef();
+
+	@Input
+	private String gitHubAccessToken;
+
+	@Input
+	private String version;
+
+	@TaskAction
+	public void createGitHubRelease() {
+		GitHubReleaseApi github = new GitHubReleaseApi(this.gitHubAccessToken);
+
+		String body = readReleaseNotes();
+		Release release = Release.tag(this.version)
+				.name(this.version)
+				.body(body)
+				.preRelease(false)
+				.build();
+		github.publishRelease(this.repository, release);
+	}
+
+	private String readReleaseNotes() {
+		Project project = getProject();
+		File inputFile = project.file(Paths.get(project.getBuildDir().getPath(), "changelog/release-notes.md"));
```

Comment:
Rather than hardcoding "changelog/release-notes.md" here, we can have a public constant in `GitHubChangelogPlugin`, for example `public static final String RELEASE_NOTES_PATH = "changelog/release-notes.md";` and access it here as `GitHubChangelogPlugin.RELEASE_NOTES_PATH`. (the variable name could be anything)